feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/presentation/main/SwapValidationFailureUi.kt

```diff
@@ -0,0 +1,228 @@
+    )
+}
+
+fun CoroutineScope.handleTooSmallRemainingBalance(
+    title: String,
+    message: String,
+    resourceManager: ResourceManager,
+    actions: ValidationFlowActions<*>,
+    amountInputMixin: SwapAmountInputMixin.Presentation
+): TransformedFailure {
+    return handleErrorToSwapMax(
+        title = title,
+        message = message,
+        resourceManager = resourceManager,
+        amountInputMixin = amountInputMixin,
+        negativeButtonText = resourceManager.getString(R.string.common_proceed),
+        clickNegativeButton = { actions.resumeFlow() }
+    )
+}
+
+fun CoroutineScope.handleErrorToSwapMax(
+    title: String,
+    message: String,
+    resourceManager: ResourceManager,
+    amountInputMixin: SwapAmountInputMixin.Presentation,
+    negativeButtonText: String,
+    clickNegativeButton: () -> Unit
+): TransformedFailure {
+    return TransformedFailure.Custom(
+        CustomDialogDisplayer.Payload(
+            title = title,
+            message = message,
+            customStyle = R.style.AccentAlertDialogTheme,
+            okAction = CustomDialogDisplayer.Payload.DialogAction(
+                title = resourceManager.getString(R.string.swap_failure_swap_max_button),
+                action = {
+                    launch {
+                        amountInputMixin.maxAction.maxClick.first()?.invoke()
+                    }
+                }
+            ),
+            cancelAction = CustomDialogDisplayer.Payload.DialogAction(
+                title = negativeButtonText,
+                action = clickNegativeButton
+            )
+        )
+    )
+}
+
+fun CoroutineScope.handleErrorToSwapMin(
+    reason: AmountOutIsTooLowToStayAboveED,
+    resourceManager: ResourceManager,
+    amountOutInputMixin: SwapAmountInputMixin.Presentation
+): TransformedFailure {
+    return TransformedFailure.Custom(
+        CustomDialogDisplayer.Payload(
+            title = resourceManager.getString(R.string.swap_too_low_amount_to_stay_abow_ed_title),
+            message = resourceManager.getString(
+                R.string.swap_too_low_amount_to_stay_abow_ed_message,
+                reason.amount.formatTokenAmount(reason.asset),
+                reason.existentialDeposit.formatTokenAmount(reason.asset),
+            ),
+            customStyle = R.style.AccentAlertDialogTheme,
+            okAction = CustomDialogDisplayer.Payload.DialogAction(
+                title = resourceManager.getString(R.string.swap_failure_swap_min_button),
+                action = {
+                    launch {
+                        amountOutInputMixin.requestFocusLiveData.sendEvent()
+                        amountOutInputMixin.inputState.value = AmountChooserMixinBase.InputState(
+                            reason.existentialDeposit.toPlainString(),
+                            initiatedByUser = true,
+                            inputKind = AmountChooserMixinBase.InputState.InputKind.REGULAR
+                        )
```

Comment:
Do you think we should move this case to mixin?

Reply:
We have a `setAmount` extension, we can add `initiatedByUser ` flag to it and use it here